internal/services/applicationinsights/application_insights_resource.go

```diff
@@ -414,6 +405,132 @@ func resourceApplicationInsightsRead(d *pluginsdk.ResourceData, meta interface{}
 	return nil
 }
 
+func resourceApplicationInsightsUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).AppInsights.ComponentsClient
+	billingClient := meta.(*clients.Client).AppInsights.BillingClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := components.NewComponentID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
```

Comment:
For the update function we should parse the ID from state rather than creating a new ID

```suggestion
	id, err := components.ParseComponentID(d.Id())
	if err != nil {
		return err
	}
```